src/containerapp/azext_containerapp/containerapp_sessionpool_decorator.py

```diff
@@ -253,12 +278,26 @@ def set_up_managed_identity(self):
             self.session_pool_def["identity"] = identity_def
 
     def set_up_dynamic_configuration(self):
-        if self.get_argument_cooldown_period_in_seconds() is None:
+        if self.get_argument_lifecycle_type() is None:
+            # Auto infer lifecycle type
+            if self.get_argument_cooldown_period_in_seconds() is not None:
+                self.set_argument_lifecycle_type("Timed")
+            elif self.get_argument_max_alive_period() is not None:
+                self.set_argument_lifecycle_type("OnContainerExit")
+            else:
+                # Default to 'Timed'
+                self.set_argument_lifecycle_type("Timed")
+
+        if self.get_argument_lifecycle_type().lower() == "timed" and self.get_argument_cooldown_period_in_seconds() is None:
             self.set_argument_cooldown_period_in_seconds(300)
 
+        if self.get_argument_lifecycle_type().lower() == "oncontainerexit" and self.get_argument_max_alive_period() is None:
+            self.set_argument_max_alive_period(3600)
```

Comment:
Do you want to set default value for `max_alive_period` to `3600` if `lifecycle_type` is "oncontainerexit"?

Reply:
Per offline discussion, we want this to only default to 3600 when lifecycle_type is `OnContainerExit`, so keep this current implementation.